wp-includes/sqlite/class-wp-sqlite-pdo-user-defined-functions.php

```diff
@@ -493,6 +494,54 @@ public function regexp( $pattern, $field ) {
 		return preg_match( $pattern, $field );
 	}
 
+	/**
+	 * Method to emulate MySQL REGEXP_REPLACE() function.
+	 *
+	 * @param string|array $pattern     Regular expression to search for (or array of strings).
+	 * @param string|array $replacement The string or an array with strings to replace.
+	 * @param string|array $field       Haystack.
+	 *
+	 * @return Array if the field parameter is an array, or a string otherwise.
+	 */
+	public function regexp_replace( $field, $pattern, $replacement ) {
+		/*
+		 * If the original query says REGEXP BINARY
+		 * the comparison is byte-by-byte and letter casing now
+		 * matters since lower- and upper-case letters have different
+		 * byte codes.
+		 *
+		 * The REGEXP function can't be easily made to accept two
+		 * parameters, so we'll have to use a hack to get around this.
+		 *
+		 * If the first character of the pattern is a null byte, we'll
+		 * remove it and make the comparison case-sensitive. This should
+		 * be reasonably safe since PHP does not allow null bytes in
+		 * regular expressions anyway.
+		 */
+
+		/* Return null if one of the required parameter is null */
+		if ( is_null( $field ) || is_null( $pattern ) || is_null( $replacement ) ) {
+			return null;
+		}
+
+		/* Return null if the pattern is empty - this changes MySQL/MariaDB behavior! */
+		if ( empty( $pattern ) ) {
+			return null;
+		}
```

Comment:
Oh! Wha is the MySQL/MariaDB behavior? to just use an empty pattern? If so, let's do that.

Reply:
It is throwing an error (because without pattern it can't work at all):
> Error in the SQL query (3685): Illegal argument to a regular expression. 

@bgrgicak was suggesting to return null instead to avoid breaking things, I assume.

Reply:
Null seemed like a better response from an error. @Zodiac1978 let's update the comment to make it clear what _this changes MySQL/MariaDB behavior_ means.